src/main/kotlin/hs/flensburg/marlin/business/api/location/entity/DetailedLocationDTO.kt

```diff
@@ -0,0 +1,27 @@
+package hs.flensburg.marlin.business.api.location.entity
+
+import hs.flensburg.marlin.database.generated.tables.pojos.Location
+import kotlinx.datetime.LocalTime
+import kotlinx.datetime.toKotlinLocalTime
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class DetailedLocationDTO(
+    val id: Long?,
+    val name: String?,
+    val description: String?,
+    val address: String?,
+    val openingTime: LocalTime?,
+    val closingTime: LocalTime?,
+    val coordinates: GeoPoint?
+)
+
+fun Location.toDetailedLocationDTO() = DetailedLocationDTO(
+    id = this.id,
+    name = this.name,
+    description = this.description,
+    address = this.address,
+    openingTime = this.openingTime?.toKotlinLocalTime(),
+    closingTime = this.closingTime?.toKotlinLocalTime(),
+    coordinates = this.coordinates
+)
```

Comment:
Ich würde diese Mapping-Funktion eher als statische Funktion in das companion object von DetailedLocationDTO legen, z. B. DetailedLocationDTO.from(location). So bleibt die Instanzierunglogik beim DTO gebündelt und das Domainmodell (Location) bleibt davon unabhängig.
```suggestion
@Serializable
data class DetailedLocationDTO(
    val id: Long?,
    val name: String?,
    val description: String?,
    val address: String?,
    val openingTime: LocalTime?,
    val closingTime: LocalTime?,
    val coordinates: GeoPoint?
) {
    companion object {
        fun fromLocation(location: Location): DetailedLocationDTO {
            return DetailedLocationDTO(
                id = location.id,
                name = location.name,
                description = location.description,
                address = location.address,
                openingTime = location.openingTime?.toKotlinLocalTime(),
                closingTime = location.closingTime?.toKotlinLocalTime(),
                coordinates = location.coordinates
            )
        }
    }
}

```